src/minimal_period.jl

```diff
@@ -0,0 +1,79 @@
+export minimal_period
+
+using LinearAlgebra: norm, dot
+
+"""
+    minimal_period(ds::DynamicalSystem, po::PeriodicOrbit; kw...) → minT_po
+
+Compute the minimal period of the periodic orbit `po` of the dynamical system `ds`.
+Return the periodic orbit `minT_po` with the minimal period. In the literature, minimal 
+period is also called prime, principal or fundamental period.
+
+## Keyword arguments
+
+* `atol = 1e-4` : After stepping the point `u0` for a time `T`, it must return to `atol` neighborhood of itself to be considered periodic.
+* `maxiter = 10` : Maximum number of Poincare map iterations. Continuous-time systems only. 
```

Comment:
What do you do if you exceed this? What is returned as the period? Should it error? It is not clear from the docs what happens.

I think this number should be much higher than 10.

Reply:
Right now I return the original period `po.T`. I think warning is better than error. I will update `maxiter` to 40.